packages/vite/src/node/plugins/css.ts

```diff
@@ -2529,7 +2529,7 @@ const makeLessWorker = (
     dir: string,
     rootFile: string,
   ) => {
-    const resolved = await resolvers.less(filename, path.join(dir, '*'))
+    const resolved = await resolvers.less(filename, rootFile)
```

Comment:
I don't think this is correct? The `filename` should always be resolved from the file that imports it, not the rootFile. The problem I'm seeing is that Less is treating `@import url()` weirdly.

With `@import url('../../less/ommer.less');` in the test, the `filename` becomes `less/ommer.less`. But with  `@import '../../less/ommer.less';`, `filename` becomes `../../less/ommer.less`.

I think we need to figure out why this happens first.

Reply:
Actually it doesn't look like the test fails in main. When Vite fails to resolve it, it'll fallback to Less to load it itself (`super.loadFile`), and it'll work again.

Reply:
The difference between `@import url('../../less/ommer.less');` and `@import '../../less/ommer.less';` is the first one gets everything in url() rebased, the second one does not get rebased.

Also what do you mean
> it doesn't look like the test fails in main.

Do you want to make vite resolve it instead of falling back to less?

The test case without the fix indeed fails in main when called with `pnpm run test-build`
<img width="1322" alt="Screenshot 2024-08-14 at 3 34 30 PM" src="https://github.com/user-attachments/assets/2d149f32-f791-490d-81e4-b3308c8f7a35">

Reply:
> The test case without the fix indeed fails in main when called with `pnpm run test-build`

My bad looks like it does. I was running the playground manually and seeing that the text was darkorange even without this PR's changes.

> Do you want to make vite resolve it instead of falling back to less?

Ideally Vite should always be able to resolve it yes.

> However after the import in line 2535 we rebased the url to the rootfile. Do you mean we should rebase the file to the current file instead of rootfile? Then next time the file name will be relative to the file that imports it instead of relative to the root file.

I think I see the issue now. I didn't expect `rebaseUrls` to be the culprit, but that function should really not rebase any urls related to `@import "";` and `@import url("")` (seems to start from https://github.com/vitejs/vite/pull/7147). The goal of `rebaseUrls` is to make sure the `url()` or `data-uri()` would still reference the correct file path as that reference will be left as is after compiling to a single file.

`@import` is different because they don't get left in that single file, their contents are bundled instead, unless it's importing an external URL, data URL, or something, but in that case you wouldn't need to rebase it either.

Maybe I'm missing something, but if what I'm getting is correct. The fix is to remove this:

https://github.com/vitejs/vite/blob/67005949999054ab3cd079890ed220bc359bcf62/packages/vite/src/node/plugins/css.ts#L2501-L2504

and update this to not capture `@import url(`: https://github.com/vitejs/vite/blob/67005949999054ab3cd079890ed220bc359bcf62/packages/vite/src/node/plugins/css.ts#L1632-L1633

I'm not sure if this would break other tests, feel free to let me know though if the issue/fix is getting overwhelming and I can help make a fix too.

Reply:
I have tested the different fix you suggested but it seems breaks the original test case from #7147